llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -5737,10 +5743,19 @@ LoopVectorizationCostModel::selectInterleaveCount(ElementCount VF,
   // the InterleaveCount as if vscale is '1', although if some information about
   // the vector is known (e.g. min vector size), we can make a better decision.
   if (BestKnownTC) {
-    MaxInterleaveCount =
-        std::min(*BestKnownTC / VF.getKnownMinValue(), MaxInterleaveCount);
-    // Make sure MaxInterleaveCount is greater than 0.
-    MaxInterleaveCount = std::max(1u, MaxInterleaveCount);
+    unsigned EstimatedVF = VF.getKnownMinValue();
+    if (VF.isScalable()) {
+      if (std::optional<unsigned> VScale = getVScaleForTuning(L, TTI))
+        EstimatedVF *= *VScale;
+    }
+    if (InterleaveSmallLoopScalarReduction || (*BestKnownTC % EstimatedVF == 0))
```

Comment:
Why the special handling for `InterleaveSmallLoopScalarReduction`? It might be good to do the check `(*BestKnownTC % EstimatedVF == 0)` only for known trip counts, as the estimated trip count is just an average and may not be a good indication if the actual trip counts are multiples of the EsimatedVF

Reply:
> Why the special handling for InterleaveSmallLoopScalarReduction?

Removed the special handling for `InterleaveSmallLoopScalarReduction`. Had kept it earlier to not change its behavior as it seemed like a special-purpose flag.